### PR TITLE
chore: change to remote workflows

### PR DESCRIPTION
**Pull request by michalby24**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| New feature     | ✔                                                                    |
| Chore            | ✔                                                                    |

Further  information:
use reusable workflows

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -3,81 +3,6 @@ name: pull_request
...
-
-    steps:
-      - name: Check out Git repository
-        uses: actions/checkout@v2
-
-      - name: build Docker image
-        run: docker build -t test-build:latest .
+  reusable_workflow_job:
+    uses: MapColonies/shared-workflows/.github/workflows/pull_request.yaml@master
+    secrets: inherit
```
Comment (AnguIar) [inline .github/workflows/pull_request.yaml:8]: Add new line

File: .github/workflows/release-on-tag-push.yaml [hunk 2]
```diff
@@ -6,33 +6,6 @@ on:
...
-      - name: Publish Release to Github
-        uses: softprops/action-gh-release@v1
-        with:
-          body_path: CHANGELOG.md
-          files: ${{ env.P_COLLECTION_FILE_NAME }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  reusable_workflow_job:
+    uses: MapColonies/shared-workflows/.github/workflows/release-on-tag-push.yaml@master
+    secrets: inherit
```
Comment (AnguIar) [inline .github/workflows/release-on-tag-push.yaml:11]: Add new line

**Review by AnguIar — CHANGES_REQUESTED** [review]
Good job :100:

File: .github/workflows/build-and-push.yaml [hunk 3]
```diff
@@ -18,57 +18,6 @@ permissions:
...
-      - name: save helm chart to local registry
-        run: |
-          helm chart save . '${{ secrets.ACR_URL }}/helm/${{ env.CHART }}:${{ env.VER }}'
-        working-directory: ./helm
-      
-      - name: publish chart to acr
-        run: |
-          helm chart push ${{ secrets.ACR_URL }}/helm/${{ env.CHART }}:${{ env.VER }}
-        working-directory: ./helm
+  reusable_workflow_job:
```
Comment (shimoncohen) [inline .github/workflows/build-and-push.yaml:21]: I think the job name should still be meaningful.
Same goes for other files.

**Review by shimoncohen — CHANGES_REQUESTED** [review]

**Review by AnguIar — APPROVED** [review]

**Pull request merged**